src/UTMSP/UTMSPMapPolygonVisuals.qml

```diff
@@ -0,0 +1,708 @@
+    }
+
+    /// Calculate the default/initial 4 sided polygon
+    function defaultPolygonVertices() {
+        // Initial polygon is inset to take 2/3rds space
+        var rect = Qt.rect(mapControl.centerViewport.x, mapControl.centerViewport.y, mapControl.centerViewport.width, mapControl.centerViewport.height)
+        rect.x += (rect.width * 0.25) / 2
+        rect.y += (rect.height * 0.25) / 2
+        rect.width *= 0.75
+        rect.height *= 0.75
+
+        var centerCoord      =  mapControl.toCoordinate(Qt.point(rect.x + (rect.width / 2), rect.y + (rect.height / 2)),   false /* clipToViewPort */)
+        var topLeftCoord     =  mapControl.toCoordinate(Qt.point(rect.x, rect.y),                                          false /* clipToViewPort */)
+        var topRightCoord    =  mapControl.toCoordinate(Qt.point(rect.x + rect.width, rect.y),                             false /* clipToViewPort */)
+        var bottomLeftCoord  =  mapControl.toCoordinate(Qt.point(rect.x, rect.y + rect.height),                            false /* clipToViewPort */)
+        var bottomRightCoord =  mapControl.toCoordinate(Qt.point(rect.x + rect.width, rect.y + rect.height),               false /* clipToViewPort */)
+
+        // Initial polygon has max width and height of 3000 meters
+        var halfWidthMeters  =  Math.min(topLeftCoord.distanceTo(topRightCoord), 3000) / 2
+        var halfHeightMeters =  Math.min(topLeftCoord.distanceTo(bottomLeftCoord), 3000) / 2
+        topLeftCoord         =  centerCoord.atDistanceAndAzimuth(halfWidthMeters, -90).atDistanceAndAzimuth(halfHeightMeters, 0)
+        topRightCoord        =  centerCoord.atDistanceAndAzimuth(halfWidthMeters, 90).atDistanceAndAzimuth(halfHeightMeters, 0)
+        bottomLeftCoord      =  centerCoord.atDistanceAndAzimuth(halfWidthMeters, -90).atDistanceAndAzimuth(halfHeightMeters, 180)
+        bottomRightCoord     =  centerCoord.atDistanceAndAzimuth(halfWidthMeters, 90).atDistanceAndAzimuth(halfHeightMeters, 180)
+        console.log([topLeftCoord, topRightCoord, bottomRightCoord, bottomLeftCoord])
+        return [topLeftCoord, topRightCoord, bottomRightCoord, bottomLeftCoord]
+    }
+
+    // Generate Automatic Geofence
+    function optimalPolygonVertices() {
+
+        var firstPoints = []
+        var lastPoints = []
+        var topCoordList = []
+        var bottomCoordList = []
+        var arrayPolygon = []
+        var newPolyCoords = []
+        var Mt = []
+        var Ct = []
+        var Mb = []
+        var Cb = []
+        var thetaList = []
+        var Xt = []
+        var Yt = []
+        var Xb = []
+        var Yb = []
+        var appendedListx =[]
+        var appendedListy =[]
+        var X_Top = []
+        var Y_Top = []
+        var X_Bottom = []
+        var Y_Bottom = []
+        var coordinateListx = []
+        var coordinateListy = []
+
+        centerX = Math.round(mapControl.fromCoordinate(missionItems.get(1).coordinate,false).x)
```

Comment:
Argument is too long and fromCoordinate and missionItems.get(1).coordinate was called twise

Use
auto centerPoint = mapControl.fromCoordinate(missionItems.get(1).coordinate,false);
centerX = Math.round(centerPoint.x)
centerX = Math.round(centerPoint.y)